perllib/Open311/GetServiceRequestUpdates.pm

```diff
@@ -184,22 +189,28 @@ sub update_comments {
 }
 
 sub comment_text_for_request {
-    my ($self, $request, $problem, $state, $external_status_code) = @_;
+    my ($self, $request, $problem, $state, $old_state,
+        $ext_code, $old_ext_code) = @_;
 
     return $request->{description} if $request->{description};
 
-    my $state_params = {
-        'me.state' => $state
-    };
-    if ($external_status_code) {
-        $state_params->{'me.external_status_code'} = $external_status_code;
-    };
-
-    if (my $template = $problem->response_templates->search({
-        auto_response => 1,
-        -or => $state_params,
-    })->first) {
-        return $template->text;
+    # Response templates are only triggered if the state/external status has changed
+    my $state_changed = defined $state && defined $old_state && $state ne $old_state;
+    my $ext_code_changed = defined $ext_code && defined $old_ext_code && $ext_code ne $old_ext_code;
```

Comment:
OTOH I can see how these could not be defined; would it be a bit nicer to have  `|| ''` when fetching the old/new external status codes, so we can assume they are strings from then on.